src/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -29,20 +28,7 @@ internal class OnBehalfOfRequest : RequestBase
         internal override async Task<AuthenticationResult> ExecuteAsync(CancellationToken cancellationToken)
         {
             await ResolveAuthorityEndpointsAsync().ConfigureAwait(false);
-
-            // look for access token in the cache first.
-            // no access token is found, then it means token does not exist
-            // or new assertion has been passed. We should not use Refresh Token
-            // for the user because the new incoming token may have updated claims
-            // like mfa etc.
```

Comment:
@jennyf19  - do you understand this comment? The caching logic here was added for a purpose, but seems to be forgotten.

Reply:
Would it mean that our guidance is wrong?